### ISSUE TITLE
[Android 1.13.2-beta]: NPE on loadBundle

### ISSUE THREAD

**Issue by overminder** [Microsoft/react-native-code-push#423]
### Context

- `react-native@0.29.0`
- `react-native-code-push@1.13.2-beta`
- `Android 4.4 and 5.1, armv7a` (Shouldn't matter..)
- `CodePush` created with `isDebugMode` off.

### Stacktrace

```
java.lang.NullPointerException
at CodePushNativeModule.loadBundleLegacy(CodePushNativeModule.java:86)
at CodePushNativeModule.loadBundle(CodePushNativeModule.java:103)
at CodePushNativeModule.restartApp(CodePushNativeModule.java:460)
...
```

[Or here on master.](https://github.com/Microsoft/react-native-code-push/blob/b665afdeb5e83fb390d9ae5bfc10507c1f6d50e5/android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java#L93)

### Reproduction

0. Ensure there is a new CodePush release in the cloud.
1. Launch the React Native based app.
2. (In the JS side) Call `CodePush.sync({ installMode: IMMEDIATE })`.
3. (In the JS side, at the same time) Launch a native activity (In our case it is a Google AdMob Interstitial).
4. Keep the native activity open. Wait until the CodePush bundle is downloaded. Once the installation finishes (and `CodePushNativeModule#restartApp` is called), the app crashes.

### Analysis

The faulty null pointer came from `ReactContextBaseJavaModule#getCurrentActivity`, which returns `ReactContext#mCurrentActivity#get`. This field will be set to null in `ReactContext#onHostPause`, which in this issue will be triggered by launching another native activity.

It's quite possible that this can't be solved without changes to the react-native repo.. Maybe we can propose to add something like `Promise<Activity> getCurrentActivityEventually()` so that we can restart the activity when it comes back.

**Comment by geof90** [comment]
@overminder are you using https://www.npmjs.com/package/react-native-admob?

**Comment by overminder** [comment]
@geof90 

Thanks for your fix! That's pretty quick..

Yes, I'm using react-native-admob. But as you might have already figured out, any kind of native activity should be able to trigger this.

**Comment by geof90** [comment]
@overminder cool! Could you do us a favour and pull the latest bits from GitHub now that I have merged it, and verify that the fix works?

**Comment by geof90** [comment]
The fix was relatively quick because you provided a ton of useful information.

**Comment by overminder** [comment]
@geof90 Yes, I can confirm that 337a03e fixes this issue. Good job!

**Comment by geof90** [comment]
Awesome! We'll publish the changes to NPM tomorrow.

**Comment by geof90** [comment]
We've published `react-native-code-push@1.13.3` to NPM.
